src/material/dialog/dialog.scss

```diff
@@ -36,7 +41,7 @@ $_emit-fallbacks: true;
   height: 100%;
   display: block;
   box-sizing: border-box;
-  max-height: inherit;
+  max-height: $mat-dialog-container-max-height;
```

Comment:
There are a couple of issues with this:
1. It needs to be `inherit`, because the `max-height` comes from a parent element.
2. Setting it here means that users can't make a dialog fullscreen.

Reply:
Reverted